.eslintrc.json

```diff
@@ -263,7 +260,7 @@
         "typescript/no-namespace": "error",
         "typescript/no-parameter-properties": "error",
         "typescript/no-triple-slash-reference": "error",
-        "typescript/no-unused-vars": "error",
+        "typescript/no-unused-vars": "off",
         "typescript/no-use-before-define": "error",
```

Comment:
Slightly off-topic, but I notice we have a few of the same rules enabled for both in general and under the `typescript/` section (e.g. `no-use-before-define`). From my experience it seems both of these can flag in TypeScript files which feels a bit unnecessarily redundant. 

Is there a need to keep both or should we prefer having one enabled over the other?

Reply:
I think this is because we have JavaScript files (`/scripts`) and the rest is TypeScript. Are you sure the JavaScript one will flag on TypeScript files too? If so we can probably disable the TS one.

Reply:
> Are you sure the JavaScript one will flag on TypeScript files too?

No, I dug a bit deeper and according to the documentation for `eslint-plugin-typescript` the `typescript/*` rules should only flag TypeScript-specific constructs. Therefore any double-reporting I've seen should probably be considered a bug in that project (and may be due to enhancements to the ESLint core via the official `typescript-eslint-parser` covering more scenarios than when the rule was originally authored).